evap/evaluation/models.py

```diff
@@ -779,24 +779,13 @@ def unpublish(self):
         self._voter_count = None
         self._participant_count = None
 
-    STATE_STR_CONVERSION = {
-        State.NEW: _("new"),
-        State.PREPARED: _("prepared"),
-        State.EDITOR_APPROVED: _("editor_approved"),
-        State.APPROVED: _("approved"),
-        State.IN_EVALUATION: _("in_evaluation"),
-        State.EVALUATED: _("evaluated"),
-        State.REVIEWED: _("reviewed"),
-        State.PUBLISHED: _("published"),
-    }
-
     @classmethod
-    def state_to_str(cls, state):
-        return cls.STATE_STR_CONVERSION[state]
+    def state_to_str(cls, state: "Evaluation.State | int") -> StrOrPromise:
+        return Evaluation.State(state).label
 
     @property
     def state_str(self):
```

Comment:
I suppose we don't need this anymore, the one usage can just do `evaluation.state.label` itself

Reply:
I believe we can't. The field `state` can be just an integer so we need to convert it into an enum before calling .label.

Reply:
We could of course call `Evaluation.State(evaluation.state).label` but I believe this is verbose enough to warrant it's own function.

But if you have a good way to make sure that `state` is always set to an enum member, this would be even better

Reply:
We could rename the database field to `_state` and add

```python
@property
def state(self):
    return Evaluation.State(self._state)
```

Not sure if we would run into some problems though. @richardebeling What do you think? It sounds like a fair change to me

Reply:
I also used this pattern and it worked quite well :+1:

Reply:
As pointed out by @fekoch today: The property has the disadvantage that using the field in django db field specifications (`evaluation__state`) becomes weird (triple underscore? Does that work with the parsing?) when using django's ORM or model bakery.

I'd be fine with all proposed solutions as long as you can still use the field with django DB field lookups.

Reply:
I would argue for just keeping the `state` field as is. I decided for keeping the `state_str()` property which already existed for the one instance where it is used.

Reply:
Ah, the ORM story is a good point. Okay then, let's keep it as is :)